tracing/opentelemetry/src/main/java/io/helidon/tracing/opentelemetry/HelidonOpenTelemetry.java

```diff
@@ -49,4 +56,51 @@ public static OpenTelemetryTracer create(OpenTelemetry telemetry, Tracer tracer,
     public static io.helidon.tracing.Span create(Span span) {
         return new OpenTelemetrySpan(span);
     }
+
+    /**
+     * Check if OpenTelemetry is present by indirect properties.
+     * This class does best explicit check if OTEL_AGENT_PRESENT_PROPERTY config property is set and uses its
+     * value to set the behaviour of OpenTelemetry producer.
+     *
+     * If the value is not explicitly set, the detector does best effort to estimate indirect means if the agent is present.
+     * This detector may stop working if OTEL changes the indirect indicators.
+     */
+    public static class AgentDetector {
```

Comment:
this is basically just a util class. It has to have private constructor and it should probably be final as well.

Reply:
ok